### PR TITLE
Attachment details documentation

### LINKED ISSUE #74: Document Media library - views/attachment/details.js
Add (and update existing) documentation to .../views/attachment/details.js. 
Part of ongoing effort to document the media library.

### PR DESCRIPTION
**Pull request by maartenleenders**
Fixes #74 

Adds documentation to details.js

### CODE REVIEW

File: src/wp-includes/js/media/views/attachment/details.js [hunk 1]
```diff
@@ -2,22 +2,18 @@ var Attachment = wp.media.view.Attachment,
...
 	tagName:   'div',
 	className: 'attachment-details',
 	template:  wp.template('attachment-details'),
 
+	/**
+	 * Overrides the attributes method in the Attachment prototype.
+	 *
+	 * @since 4.2.0
+	 *
+	 * @returns {Object}
```
Comment (boblinthorst) [inline src/wp-includes/js/media/views/attachment/details.js]: Missing description of the return.

File: src/wp-includes/js/media/views/attachment/details.js [hunk 2]
```diff
@@ -2,22 +2,18 @@ var Attachment = wp.media.view.Attachment,
...
- * @augments wp.Backbone.View
- * @augments Backbone.View
- */
 Details = Attachment.extend(/** @lends wp.media.view.Attachment.Details.prototype */{
 	tagName:   'div',
 	className: 'attachment-details',
 	template:  wp.template('attachment-details'),
 
+	/**
+	 * Overrides the attributes method in the Attachment prototype.
```
Comment (boblinthorst) [inline src/wp-includes/js/media/views/attachment/details.js]: Only describes that attributes is overridden, not what the new attributes method is actually doing.

File: src/wp-includes/js/media/views/attachment/details.js [hunk 3]
```diff
@@ -60,7 +73,15 @@ Details = Attachment.extend(/** @lends wp.media.view.Attachment.Details.prototyp
...
+	 * Deletes an attachment after asking for confirmation. After deletion, keeps focus in the modal.
+	 *
+	 * @since 4.2.0
+	 *
+	 * @param {MouseEvent} event A click event.
+	 *
+	 * @returns {void}
 	 */
 	deleteAttachment: function( event ) {
 		event.preventDefault();
```
Comment (boblinthorst) [inline src/wp-includes/js/media/views/attachment/details.js]: About lines 91-92: please make this a multiline comment using `/* */`

File: src/wp-includes/js/media/views/attachment/details.js [hunk 4]
```diff
@@ -60,7 +73,15 @@ Details = Attachment.extend(/** @lends wp.media.view.Attachment.Details.prototyp
 		}
 	},
 	/**
-	 * @param {Object} event
+	 * Deletes an attachment.
+	 *
+	 * Deletes an attachment after asking for confirmation. After deletion, keeps focus in the modal.
```
Comment (boblinthorst) [inline src/wp-includes/js/media/views/attachment/details.js]: Please check line length

File: src/wp-includes/js/media/views/attachment/details.js [hunk 5]
```diff
@@ -73,7 +94,15 @@ Details = Attachment.extend(/** @lends wp.media.view.Attachment.Details.prototyp
 		}
 	},
 	/**
-	 * @param {Object} event
+	 * Sets the trash state on an attachment, or destroys it.
```
Comment (boblinthorst) [inline src/wp-includes/js/media/views/attachment/details.js]: Is 'it'  the trash state or an attachment?

File: src/wp-includes/js/media/views/attachment/details.js [hunk 6]
```diff
@@ -73,7 +94,15 @@ Details = Attachment.extend(/** @lends wp.media.view.Attachment.Details.prototyp
...
+	 *
+	 * If the mediaTrash setting is set to true, trashes the attachment. Otherwise, it destroys it.
```
Comment (boblinthorst) [inline src/wp-includes/js/media/views/attachment/details.js]: What's  the second 'it'?

File: src/wp-includes/js/media/views/attachment/details.js [hunk 7]
```diff
@@ -116,11 +157,19 @@ Details = Attachment.extend(/** @lends wp.media.view.Attachment.Details.prototyp
 			this.$el.addClass('needs-refresh');
 		}
 	},
+
 	/**
-	 * When reverse tabbing(shift+tab) out of the right details panel, deliver
+	 * When reverse tabbing(shift+tab) out of the right details panel, delivers
```
Comment (boblinthorst) [inline src/wp-includes/js/media/views/attachment/details.js]: space after 'tabbing'

File: src/wp-includes/js/media/views/attachment/details.js [hunk 8]
```diff
@@ -116,11 +157,19 @@ Details = Attachment.extend(/** @lends wp.media.view.Attachment.Details.prototyp
...
 	 *
-	 * @param {Object} event
+	 * @since 4.2.0
+	 *
+	 * @fires wp.media.controller.MediaLibrary#attachment:details:shift-tab
+	 * @fires wp.media.controller.MediaLibrary#attachment:keydown:arrow
+	 *
+	 * @param {KeyboardEvent} event A keyboard event.
+	 *
+	 * @returns {boolean|void}
```
Comment (boblinthorst) [inline src/wp-includes/js/media/views/attachment/details.js]: Missing return description in case

**Comment by atimmer** [comment]
Committed in https://core.trac.wordpress.org/changeset/45561.

**Pull request closed without merging**